src/io/types/CANMessage.cpp

```diff
@@ -0,0 +1,46 @@
+#include <stdint.h>
+#include <cstring>
+
+#include <EVT/io/types/CANMessage.hpp>
+
+namespace EVT::core::IO {
+
+CANMessage::CANMessage(uint32_t id, uint8_t dataLength, uint8_t* payload) {
+    this->id = id;
+    this->dataLength = dataLength;
+
+    // Copy contents of provided payload into message's payload
+    for (int i = 0; i < dataLength; i++)
```

Comment:
Upper bounds check on dataLength being larger than 8 bytes?

Reply:
Yeah that should have bound checking. In the future we should have either an assertion or other way to notify the user of an issue, but for now I'll just bound the dataLength to the max frame size.